src/data.rs

```diff
@@ -10,8 +10,9 @@ use crate::dev::Payload;
 use crate::extract::FromRequest;
 use crate::request::HttpRequest;
 
-/// Application data factory
+/// Data factory.
 pub(crate) trait DataFactory {
+    /// Return true if modifications were made to extensions map.
     fn create(&self, extensions: &mut Extensions) -> bool;
```

Comment:
I think with the new behavior the boolean return is not needed anymore.

We could also get rid of this trait and return a boxed function directly in App::data_factory like Box<dy Fn<&mut Extensions>>.

Reply:
yep just documented what seemed to be the behavior on the way, i agree this should be removed next, only didn't because of data factory as you pointed out